### PR TITLE
Add a macro to span functions and methods

### PR DESCRIPTION
**Pull request by sfackler**
This is a nicer API than manually creating spans at the start of
functions, particularly for async methods.

### CODE REVIEW

**Comment by sfackler** [comment]
We can/should expand this to allow you to add tags/etc, but that can happen later.

File: zipkin-macros/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+/// Both normal and `async` methods and functions are supported. The name of the span is specified as an argument
+/// to the macro attribute.
+///
+/// Requires the `macros` Cargo feature.
+///
+/// # Examples
+///
+/// ```ignore
+/// #[zipkin::spanned(name = "shave yaks")]
+/// fn shave_some_yaks(&[Yak]) {
```
Comment (dtobin) [inline zipkin-macros/src/lib.rs]:
```suggestion
/// fn shave_some_yaks(yaks: &[Yak]) {
```

Comment (sfackler) [reply zipkin-macros/src/lib.rs]: Fixed

**Comment by sfackler** [comment]
Yeah, it's an implementation detail.

**Pull request merged**